### PR TITLE
Post build fn

### PR DESCRIPTION
**Pull request by Sohalt**

### CODE REVIEW

File: src/nextjournal/clerk/builder.clj [hunk 1]
```diff
@@ -131,18 +131,15 @@
...
                      paths
                      (throw (ex-info "`:paths` must be sequential" {:paths paths})))
-             paths-fn (if (qualified-symbol? paths-fn)
-                        (try
-                          (if-let [resolved-var  (requiring-resolve paths-fn)]
-                            (let [resolved-paths (cond-> @resolved-var
+             paths-fn (if paths-fn
+                        (do (when-not (qualified-symbol? paths-fn)
+                              (throw (ex-info "`:path-fn` must be a qualified symbol pointing at an existing var." {:paths-fn paths-fn})))
+                            (let [resolved-var (resolve-var paths-fn)
```
Comment (Sohalt) [inline src/nextjournal/clerk/builder.clj:137]: resolve-var needs to be defined before

Comment (mk) [reply src/nextjournal/clerk/builder.clj:145]: thanks, shouldn't push without running `clj -X:test`

File: src/nextjournal/clerk/builder.clj [hunk 2]
```diff
@@ -131,18 +131,15 @@
   (->> (cond paths (if (sequential? paths)
                      paths
                      (throw (ex-info "`:paths` must be sequential" {:paths paths})))
-             paths-fn (if (qualified-symbol? paths-fn)
-                        (try
-                          (if-let [resolved-var  (requiring-resolve paths-fn)]
-                            (let [resolved-paths (cond-> @resolved-var
+             paths-fn (if paths-fn
+                        (do (when-not (qualified-symbol? paths-fn)
```
[the remark this answers is not in the document]
Comment (mk) [reply src/nextjournal/clerk/builder.clj:142]: yeah I redid it here to have a more accurate error…

[the remark this answers is not in the document]
Comment (Sohalt) [reply src/nextjournal/clerk/builder.clj:142]: but the error is almost the same, no?
Also the error message now says `:path-fn` (singular) whereas the `:keys` destructuring looks for `paths-fn` (plural).
I previously passed the key to `resolve-fn` to make sure the key mentioned in the error message is the same that's looked up in the options.

**Comment by mk** [comment]
Let's revisit this when we have a concrete use case.

**Pull request closed without merging**